File: Software Technologies/Java Basics - Exercises/src/PhonebookUpgrade.java
import java.util.Scanner;
import java.util.TreeMap;

public class PhonebookUpgrade {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        TreeMap<String, String> sortedPhonebook = new TreeMap<>();
        String input = scanner.nextLine();

        while (true) {
            if (input.equals("END")) {
                break;
            }
            String[] listOfAddedNameAndNumber = input.split(" ");
            String command = listOfAddedNameAndNumber[0];
            if (command.equals("ListAll")){
                for (String item : sortedPhonebook.keySet()) {
                    System.out.println(item + " -> " + sortedPhonebook.get(item));
                }
            }else{
                String nameOfInput = listOfAddedNameAndNumber[1];

                switch (command) {
                    case "A":
                        String numberOfInput = listOfAddedNameAndNumber[2];

                        if (!sortedPhonebook.containsKey(nameOfInput)){
                            sortedPhonebook.put(nameOfInput, numberOfInput);
                        }else{
                            sortedPhonebook.replace(nameOfInput, numberOfInput);
                        }

                        break;
                    case "S":
                        if (sortedPhonebook.containsKey(nameOfInput)){
                            for (String item: sortedPhonebook.keySet()) {
                                item = nameOfInput;
                                System.out.println(item + " -> " + sortedPhonebook.get(item));
                                break;
                            }
                        }else{
                            System.out.println("Contact " + nameOfInput + " does not exist.");
                        }
                        break;
                        default:
                            break;
                }
            }

            input = scanner.nextLine();
        }
    }
}

File: ProgrammingFundamentals/14.DictionariesLambdaAndLINQ-Exercises/02.PhonebookUpgrade/PhonebookUpgrade.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace _02.PhonebookUpgrade
{
    class PhonebookUpgrade
    {
        static void Main(string[] args)
        {
            var input = Console.ReadLine();

            var phonebook = new SortedDictionary<string, string>();

            while (input != "END")
            {
                var inputTokens = input.Split(' ').ToList();

                var command = inputTokens[0];

                if (command == "A")
                {
                    var name = inputTokens[1];
                    var phoneNumber = inputTokens[2];

                    phonebook[name] = phoneNumber;
                }
                else if (command == "S")
                {
                    var name = inputTokens[1];
                    if (phonebook.ContainsKey(name))
                    {
                        Console.WriteLine(name + " -> " + phonebook[name]);
                    }
                    else
                    {
                        Console.WriteLine($"Contact {name} does not exist.");
                    }
                }
                else if (command == "ListAll")
                {
                    foreach (var contact in phonebook)
                    {
                        Console.WriteLine($"{contact.Key} -> {contact.Value}");
                    }
                }

                input = Console.ReadLine();
            }
        }
    }
}
